epp-client/src/domain/transfer.rs

```diff
@@ -4,466 +4,90 @@ use epp_client_macros::*;
-///
-/// use epp_client::config::{EppClientConfig, RegistryConfig};
-/// use epp_client::EppClient;
-/// use epp_client::domain::transfer::DomainTransferReject;
-/// use epp_client::common::NoExtension;
-/// use epp_client::login::Login;
-/// use epp_client::logout::Logout;
-///
-/// #[tokio::main]
-/// async fn main() {
-///     // Create a config
-///     let mut registry: HashMap<String, RegistryConfig> = HashMap::new();
-///     registry.insert(
-///         "registry_name".to_owned(),
-///         RegistryConfig {
-///             host: "example.com".to_owned(),
-///             port: 700,
-///             tls_files: None,
-///         },
-///     );
-///     let config = EppClientConfig { registry };
-///
-///     // Create an instance of EppClient, passing the config and the registry you want to connect to
-///     let mut client = match EppClient::new(&config, "registry_name").await {
-///         Ok(client) => client,
-///         Err(e) => panic!("Failed to create EppClient: {}",  e)
-///     };
-///
-///     let login = Login::<NoExtension>::new("username", "password", None);
-///     client.transact(login, "transaction-id").await.unwrap();
-///
-///     // Create an DomainTransferReject instance
-///     let domain_transfer_reject = DomainTransferReject::<NoExtension>::new(
-///         "eppdev-100.net"
-///     );
-///
-///     // send it to the registry and receive a response of type DomainTransferRejectResponse
-///     let response = client.transact(domain_transfer_reject, "transaction-id").await.unwrap();
-///
-///     println!("{:?}", response);
-///
-///     let logout = Logout::<NoExtension>::new();
-///     client.transact(logout, "transaction-id").await.unwrap();
-/// }
-/// ```
-impl<E: EppExtension> DomainTransferReject<E> {
-    pub fn new(name: &str) -> DomainTransferReject<NoExtension> {
-        DomainTransferReject {
-            request: DomainTransferReq {
-                operation: "reject".to_string(),
-                domain: DomainTransferReqData {
-                    xmlns: XMLNS.to_string(),
-                    name: name.into(),
-                    period: None,
-                    auth_info: None,
-                },
-            },
-            extension: None,
-        }
-    }
+impl Transaction<NoExtension> for DomainTransferUpdate {
+    type Response = ();
+    type ExtensionResponse = NoExtension;
+}
 
-    pub fn with_extension<F: EppExtension>(self, extension: F) -> DomainTransferReject<F> {
-        DomainTransferReject {
-            request: self.request,
-            extension: Some(extension),
-        }
-    }
+#[derive(Serialize, Deserialize, Debug, ElementName)]
+#[element_name(name = "transfer")]
+pub struct DomainTransferUpdate {
+    /// The transfer operation to perform indicated by the 'op' attr
+    /// The values are one of approve, reject, cancel
+    #[serde(rename = "op")]
+    operation: String,
+    #[serde(rename = "domain:transfer")]
+    domain: DomainTransferReqData,
 }
```

Comment:
Isn't this struct a duplicate of `DomainTransferRequest`?

Reply:
Yes, but it has a different Transaction impls. We can probably deduplicate the inner type somehow  though.

Reply:
Your changes in #43 changed the response type for all the domain transfer commands to be the same `DomainTransferResponse`, so I was able to fold `DomainTransferRequest`/`DomainTransferUpdate` back into a single type (now called `DomainTransfer` to align with the other command types).